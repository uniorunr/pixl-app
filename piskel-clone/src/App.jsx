import React, { Component, Fragment } from 'react';
import ReactDOM from 'react-dom';
import NavBar from './components/NavBar/NavBar';
import Main from './components/Main/Main';
import './App.scss';

class App extends Component {
  constructor() {
    super();

    this.state = {};
  }

  render() {
    return (
      <Fragment>
        <NavBar />
        <Main />
      </Fragment>
    );
  }
}

ReactDOM.render(<App />, document.getElementById('root'));
