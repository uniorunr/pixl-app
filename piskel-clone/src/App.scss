@import url('https://fonts.googleapis.com/css?family=Roboto:400,500,700&display=swap');

html {
  height: 100%;

  body {
    height: 100%;
    margin: 0;
    font-size: 10px;
    font-family: "Roboto", "Arial", sans-serif;
  }
}


