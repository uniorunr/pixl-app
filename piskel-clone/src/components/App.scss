@import url('https://fonts.googleapis.com/css?family=Roboto:400,500,700&display=swap');

::-webkit-scrollbar {
  width: 10px;
}

::-webkit-scrollbar-track {
  background: #f1f1f1;
}

::-webkit-scrollbar-thumb {
  background: #101010;
}

::-webkit-scrollbar-thumb:hover {
  background: black;
}

html {
  height: 100%;
  font-size: 10px;

  body {
    height: 100%;
    margin: 0;
    font-family: "Roboto", "Arial", sans-serif;
    display: flex;
    flex-direction: column;
    overflow: hidden;

    .root {
      display: flex;
      flex-direction: column;
      overflow: hidden;
    }

    .visually-hidden {
      position: absolute;

      width: 1px;
      height: 1px;
      margin: -1px;

      clip: rect(0 0 0 0);
    }
  }
}


