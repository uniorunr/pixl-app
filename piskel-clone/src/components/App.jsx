import React, { Component, Fragment } from 'react';
import * as firebase from 'firebase/app';
import 'firebase/auth';
import NavBar from './NavBar/NavBar';
import Main from './Main/Main';
import './App.scss';
import FireBase from '../firebase/firebase';
import toolsDataJSON from './toolsData.json';

FireBase.init();

class App extends Component {
  state = {
    userData: null,
    signInState: null,
    toolsData:
      JSON.parse(sessionStorage.getItem('toolsData'))
      || JSON.parse(JSON.stringify(toolsDataJSON)),
    currToolId: 'pen',
  };

  updateCurrentTool = (tool) => {
    this.setState({
      currToolId: tool,
    });
  };

  updateToolsData = (data) => {
    this.setState({
      toolsData: data,
    });
  };

  componentDidMount = () => {
    firebase.auth().onAuthStateChanged((user) => {
      if (user) {
        this.setState({
          userData: user,
          signInState: null,
        });
      }
    });
  };

  updateSignInState = (state) => {
    this.setState({
      signInState: state,
    });
  };

  render() {
    const {
      userData, signInState, toolsData, currToolId,
    } = this.state;

    return (
      <Fragment>
        <NavBar
          userData={userData}
          updateSignInState={this.updateSignInState}
          signInState={signInState}
          toolsData={toolsData}
          updateCurrentTool={this.updateCurrentTool}
          updateToolsData={this.updateToolsData}
        />
        <Main
          toolsData={toolsData}
          currToolId={currToolId}
          updateCurrentTool={this.updateCurrentTool}
        />
      </Fragment>
    );
  }
}

export default App;
