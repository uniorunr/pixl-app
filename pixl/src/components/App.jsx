import React, { Component, Fragment } from 'react';
import * as firebase from 'firebase/app';
import 'firebase/auth';
import NavBar from './NavBar/NavBar';
import Main from './Main/Main';
import LandingPage from './LandingPage/LandingPage';
import './App.scss';
import FireBase from '../firebase/firebase';
import appDataJSON from './appData.json';

FireBase.init();

class App extends Component {
  state = {
    userData: null,
    signInState: null,
    section: sessionStorage.getItem('section') || 'landing',
    toolsData:
      JSON.parse(sessionStorage.getItem('toolsData'))
      || JSON.parse(JSON.stringify(appDataJSON)).tools,
    currToolId: 'pen',
    framesShortcuts:
      JSON.parse(sessionStorage.getItem('framesShortcuts'))
      || JSON.parse(JSON.stringify(appDataJSON)).frames,
    layersShortcuts:
      JSON.parse(sessionStorage.getItem('layersShortcuts'))
      || JSON.parse(JSON.stringify(appDataJSON)).layers,
  };

  toggleSection = (state) => {
    this.setState({
      section: state,
    });
  };

  updateCurrentTool = (tool) => {
    this.setState({
      currToolId: tool,
    });
  };

  updateToolsData = (data) => {
    this.setState({
      toolsData: data,
    });
  };

  updateFrameShortcuts = (data) => {
    this.setState({
      framesShortcuts: data,
    });
  };

  updateLayersShortcuts = (data) => {
    this.setState({
      layersShortcuts: data,
    });
  };

  componentDidMount = () => {
    firebase.auth().onAuthStateChanged((user) => {
      if (user) {
        this.setState({
          userData: user,
          signInState: null,
        });
      }
    });
  };

  updateSignInState = (state) => {
    this.setState({
      signInState: state,
    });
  };

  render() {
    const {
      userData,
      signInState,
      toolsData,
      currToolId,
      framesShortcuts,
      layersShortcuts,
      section,
    } = this.state;

    return (
      <Fragment>
        <NavBar
          userData={userData}
          updateSignInState={this.updateSignInState}
          signInState={signInState}
          toolsData={toolsData}
          framesShortcuts={framesShortcuts}
          layersShortcuts={layersShortcuts}
          updateCurrentTool={this.updateCurrentTool}
          updateToolsData={this.updateToolsData}
          updateFrameShortcuts={this.updateFrameShortcuts}
          updateLayersShortcuts={this.updateLayersShortcuts}
          toggleSection={this.toggleSection}
        />
        {section === 'landing' ? (
          <LandingPage toggleSection={this.toggleSection} />
        ) : (
          <Main
            toolsData={toolsData}
            currToolId={currToolId}
            updateCurrentTool={this.updateCurrentTool}
            framesShortcuts={framesShortcuts}
            layersShortcuts={layersShortcuts}
          />
        )}
      </Fragment>
    );
  }
}

export default App;
