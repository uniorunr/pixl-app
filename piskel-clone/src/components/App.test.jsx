import React from 'react';
import renderer from 'react-test-renderer';
import { JSDOM } from 'jsdom';
import App from './App';

const dom = new JSDOM();
global.document = dom.window.document;
global.window = dom.window;

const canvas = document.createElement('canvas');
canvas.setAttribute('id', 'canvas');
document.body.appendChild(canvas);

it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
