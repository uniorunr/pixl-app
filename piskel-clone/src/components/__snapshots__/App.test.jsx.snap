// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="navbar"
  >
    <nav
      className="navbar__content"
    >
      <h1
        className="navbar__logo"
      >
        <a
          href="/"
        >
          <img
            alt="Pixl logo"
            src="test-file-stub"
          />
          <span>
            Pixl!
          </span>
        </a>
      </h1>
      <button
        className="navbar__sign-in-button"
        type="button"
      >
        <span>
          Sign in
        </span>
      </button>
    </nav>
  </div>,
  <main
    className="main"
  >
    <section
      className="tools-section"
    >
      <h2
        className="visually-hidden"
      >
        Tools Section
      </h2>
      <div
        className="tools"
      >
        <button
          className="tools__pen tools_state_active"
          data-listener="true"
          id="pen"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <i
            className="fas fa-pen"
          />
        </button>
        <button
          className="tools__paint-bucket"
          data-listener="true"
          id="paint-bucket"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <i
            className="fas fa-fill-drip"
          />
        </button>
        <button
          className="tools__choose-color"
          data-listener="true"
          id="choose-color"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <i
            className="fas fa-eye-dropper"
          />
        </button>
        <button
          className="tools__eraser"
          data-listener="true"
          id="eraser"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <i
            className="fas fa-eraser"
          />
        </button>
      </div>
    </section>
    <section
      className="canvas-section"
    >
      <canvas
        className="canvas-section__canvas"
        height={512}
        id="canvas"
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
        width={512}
      />
    </section>
    <section
      className="preview-section"
    >
      <div
        className="preview"
      >
        <canvas
          className="preview__canvas"
          height="150"
          id="preview-canvas"
          width="150"
        />
        <button
          className="preview__fullscreen-button"
          type="button"
        >
          <i
            className="fas fa-expand"
          />
        </button>
        <div
          className="fps-control"
        >
          <label
            htmlFor="fps-control__input"
          >
            <span />
            <span>
              FPS
            </span>
            <input
              className="fps-control__input"
              id="fps-control__input"
              max="24"
              min="1"
              name="volume"
              step="1"
              type="range"
            />
          </label>
        </div>
        <form
          className="canvas-size-controls"
          name="canvas-size-controls"
        >
          <label
            htmlFor="canvas-32"
          >
            32
            <input
              defaultChecked={true}
              id="canvas-32"
              name="radio"
              onChange={[Function]}
              type="radio"
              value="32"
            />
          </label>
          <label
            htmlFor="canvas-64"
          >
            64
            <input
              id="canvas-64"
              name="radio"
              onChange={[Function]}
              type="radio"
              value="64"
            />
          </label>
          <label
            htmlFor="canvas-128"
          >
            128
            <input
              id="canvas-128"
              name="radio"
              onChange={[Function]}
              type="radio"
              value="128"
            />
          </label>
        </form>
      </div>
    </section>
  </main>,
]
`;
